source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/lib/index.ts

```diff
@@ -90,14 +110,42 @@ export class LambdaToElasticSearchAndKibana extends Construct {
     super(scope, id);
     defaults.CheckProps(props);
 
+    if (props.vpcProps && !props.deployVpc) {
+      throw new Error("Error - deployVpc must be true when defining vpcProps");
+    }
+
+    if (props.lambdaFunctionProps?.vpc || props.lambdaFunctionProps?.vpcSubnets) {
+      throw new Error("Error - Define VPC using construct parameters not Lambda function props");
+    }
+
+    if (props.esDomainProps?.vpcOptions) {
+      throw new Error("Error - Define VPC using construct parameters not Elasticsearch props");
+    }
+
+    if (props.deployVpc || props.existingVpc) {
+      this.vpc = defaults.buildVpc(scope, {
+        defaultVpcProps: defaults.DefaultIsolatedVpcProps(),
+        existingVpc: props.existingVpc,
+        userVpcProps: props.vpcProps,
+        constructVpcProps: {
+          enableDnsHostnames: true,
+          enableDnsSupport: true,
+        },
+      });
+    }
+
     this.lambdaFunction = defaults.buildLambdaFunction(this, {
       existingLambdaObj: props.existingLambdaObj,
-      lambdaFunctionProps: props.lambdaFunctionProps
+      lambdaFunctionProps: props.lambdaFunctionProps,
+      vpc: this.vpc
     });
 
     // Find the lambda service Role ARN
     const lambdaFunctionRoleARN = this.lambdaFunction.role?.roleArn;
 
+    const zoneProps: elasticsearch.CfnDomainProps = defaults.checkMultiAvailabilityZoneSupport(this.lambdaFunction, this.vpc, props.esDomainProps);
```

Comment:
Not clear what is going on here - we're checking Multiple AZ support for what?

Is this also where vpc is provided to the ES domain? If so, that's kind of hidden within a method that is supposedly just checking something.